ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -549,8 +562,13 @@ KEXT_STATIC int HandleVnodeOperation(
         {
             if (FileFlagsBitIsSet(currentVnodeFileFlags, FileFlags_IsEmpty))
             {
-                // Prevent system services from hydrating files as this tends to cause deadlocks with the kauth listeners for Antivirus software
-                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, CallbackPolicy_UserInitiatedOnly, &root, &vnodeFsidInode, &kauthResult, kauthError))
+                bool isWriteOperation = ActionBitIsSet(action, KAUTH_VNODE_WRITE_ATTRIBUTES | KAUTH_VNODE_WRITE_EXTATTRIBUTES | KAUTH_VNODE_WRITE_DATA | KAUTH_VNODE_APPEND_DATA);
+                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid,
+                    // Prevent system services from hydrating files as this tends to cause deadlocks with the kauth listeners for Antivirus software
+                    CallbackPolicy_UserInitiatedOnly,
+                    // Prevent write access to empty files in offline roots. For now allow reads, and always allow the user to delete files.
+                    isWriteOperation,
+                    &root, &vnodeFsidInode, &kauthResult, kauthError))
```

Comment:
nit: Can we have all the params on their own line for readability?

Reply:
Done, for all calls to this function.